test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -36,54 +35,13 @@ trait RoleSpecHelper {
 
     // pre-conditions
     if (cfg.role != "*") {
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "Spark role should exist.")
-    }
-
-    val expectedUsedCpus = {
-      val m = MesosCluster.loadStates(mesosConsoleUrl)
-      val tmp = m.slaves.map { x => x.resources.cpu }.sum
-      if (isInClusterMode) {
-          tmp - 1 // minus the cpus owned by the Spark Cluster, which is started before the tests in cluster mode
-      }
-      else tmp
+      assert(m.slaves.flatMap { x => x.reservedResources.keys }.contains(cfg.role), "Spark role should exist.")
     }
 
-    val mesosUrl = mesosConsoleUrl
-    val accum = sc.accumulator[Double](0L, "cpuCounter")
-    val partitions = 40 // give it enough tasks
-
-    // use enough tasks to utilize all expected resources for tested role and *, pick a task in the middle and observe resources
-    val rdd = sc.makeRDD(1 to 100, partitions).mapPartitionsWithIndex{ (idx, iterator) =>
-
-      var counter = 1
-      val retries = 1000
-      val partitionNumber = partitions / 2
-      val timeToSleep = 10 // small resolution to catch the event of a fully utilized cluster
-      var max = 0
-
-      if (idx == partitionNumber) {
-        while (counter <= retries) {
-          val currentNumOfCpus = MesosCluster.loadStates(mesosUrl).sparkFramework.get.resources.cpu
-
-          if (max <= currentNumOfCpus){ max = currentNumOfCpus.toInt }
-
-          if (currentNumOfCpus != expectedUsedCpus) {
-            Thread.sleep(timeToSleep)
-            counter += 1
-          } else {
-            counter = retries + 1 // exit
-          }
-        }
-
-        if (idx == partitionNumber) {
-          accum += max
-        }
-      }
-      iterator
-    }
+    assert(m.sparkFramework.get.role == cfg.role)
 
+    val rdd = sc.makeRDD(1 to 5)
     val res = rdd.sum()
-    assert(accum.value == expectedUsedCpus, "All cpu resources should be utilized.")
```

Comment:
> What you need is to get only the role resource offers and make sure that is used as expected and you can assert it.

If I'm understanding you correctly, this won't work in DCOS.  `cfg.role` in the DCOS tests is (*).  However, any assertion that the driver is using all unreserved (*) CPUs would fail, because there are other frameworks using unreserved resources.  Namely, Marathon is running the dispatcher, which is using 1 CPU on (*), and the dispatcher is running the driver, which is using 1 CPU on (*).

If there were someway for me to compute available = (total - used), then I could assert that the CPUs used by spark equal this number, but there's no way to do that in Mesos.  state.json provides the the total, but there's no endpoint that provides "used".

> Yet it does test it implicitly...

It's testing that the driver framework is launching tasks on resources reserved for the role.  That's a nice property to test, but it's purely testing Mesos, not the Spark+Mesos integration, which is the domain of this test suite.  If we merely test that the framework has registered with the expected role (`cfg.role`), then we can assume that Mesos is functioning correctly and offering resources to that framework.  I would be fine with testing that we actually launch tasks on those resources, but AFAIK, there's simply no way.